### PR TITLE
Support for common radial gradient fills.

### PR DESCRIPTION
**Pull request by simeoncran**

### CODE REVIEW

File: source/WinCompData/CompositionRadialGradientBrush.cs [hunk 1]
```diff
@@ -15,10 +15,13 @@ internal CompositionRadialGradientBrush()
         {
         }
 
+        // Default is 0.5, 0.5
         public Vector2? EllipseCenter { get; set; }
 
+        // Default is 0.5, 0.5
         public Vector2? EllipseRadius { get; set; }
```
Comment (eliezerpMS) [inline source/WinCompData/CompositionRadialGradientBrush.cs]: >EllipseRadius [](start = 24, length = 13)

Just curious but MSDN says the default is (1,1)?

Comment (eliezerpMS) [reply source/WinCompData/CompositionRadialGradientBrush.cs]: MSDN says GradientOriginOffset default is (0.5, 0.5)

---
In reply to: [321944773](https://github.com/windows-toolkit/Lottie-Windows/pull/152#discussion_r321944773) [](ancestors = 321944773)

File: source/LottieToWinComp/LottieToWinCompTranslator.cs [hunk 2]
```diff
@@ -1058,28 +1058,12 @@ internal void UpdateFromStack(Stack<ShapeLayerContent> stack)
...
-                            break;
-
                         case ShapeContentType.LinearGradientStroke:
                         case ShapeContentType.RadialGradientStroke:
                             _owner._issues.GradientStrokeIsNotSupported();
                             break;
 
+                        case ShapeContentType.RadialGradientFill:
                         case ShapeContentType.LinearGradientFill:
                         case ShapeContentType.SolidColorFill:
```
Comment (eliezerpMS) [inline source/LottieToWinComp/LottieToWinCompTranslator.cs]: Super nit: I would not mind this to be alphabetized

File: source/LottieToWinComp/LottieToWinCompTranslator.cs [hunk 3]
```diff
@@ -2662,6 +2646,113 @@ CompositionLinearGradientBrush TranslateLinearGradientFill(TranslationContext co
...
+                // We don't yet support opacity stops.
+                _issues.GradientFillIsNotSupported();
+            }
+
+            var opacityPercentValue = opacityPercent.InitialValue;
+
+            var result = _c.CreateRadialGradientBrush();
+
+            // BodyMovin specifies start and end points in absolute values.
+            result.MappingMode = CompositionMappingMode.Absolute;
```
Comment (eliezerpMS) [inline source/LottieToWinComp/LottieToWinCompTranslator.cs]: Does it make sense to encode this in the CompositionRadialGradientBrush to keep the translator agnostic of what the parser actually had to parse?

File: source/LottieToWinComp/LottieToWinCompTranslator.cs [hunk 4]
```diff
@@ -2662,6 +2646,113 @@ CompositionLinearGradientBrush TranslateLinearGradientFill(TranslationContext co
             return result;
         }
 
+        CompositionRadialGradientBrush TranslateRadialGradientFill(TranslationContext context, RadialGradientFill shapeFill, TrimmedAnimatable<double> opacityPercent)
```
Comment (eliezerpMS) [inline source/LottieToWinComp/LottieToWinCompTranslator.cs]: > CompositionRadialGradientBrush TranslateRadialGradientFill(TranslationContext context, RadialGradientFill shapeFill, TrimmedAnimatable<double> opacityPercent) [](start = 7, length = 159)

Nit: maybe make this line shorter

File: source/WinCompData/CompositionRadialGradientBrush.cs (hunk 1 above)
Comment (simeoncran) [reply source/WinCompData/CompositionRadialGradientBrush.cs]: MSDN is not consistent with reality. I created a pair of PRs to fix this in the docs. E.g. [https://github.com/MicrosoftDocs/winrt-api/pull/1217](https://github.com/MicrosoftDocs/winrt-api/pull/1217)

---
In reply to: [321944893](https://github.com/windows-toolkit/Lottie-Windows/pull/152#discussion_r321944893) [](ancestors = 321944893,321944773)

File: source/LottieToWinComp/LottieToWinCompTranslator.cs (hunk 3 above)
Comment (simeoncran) [reply source/LottieToWinComp/LottieToWinCompTranslator.cs]: Discussed in person.

---
In reply to: [321945409](https://github.com/windows-toolkit/Lottie-Windows/pull/152#discussion_r321945409) [](ancestors = 321945409)

**Pull request merged**